packages/database/src/migrations/20210120051444-ReSyncSubmissionsAfterRelease70-modifies-data.js

```diff
@@ -0,0 +1,121 @@
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const MIN_SURVEY_END_TIME = '2021-01-14';
+const SYNC_QUEUE_KEY = 'dhisSyncQueue';
+const CHANGE_BATCH_SIZE = 1000;
+
+const resolveAfterXChanges = (changeChannel, resolve, x) => {
+  if (x === 0) {
+    resolve();
+    return;
+  }
+  let changeCount = 0;
+  changeChannel.addChangeHandler(() => {
+    changeCount++;
+    if (changeCount === x) resolve();
+  });
+};
+
+const markRecordsForResync = async (changeChannel, recordType, records) => {
+  for (let batchStartIndex = 0; batchStartIndex < records.length; batchStartIndex += 1000) {
+    await new Promise(resolve => {
+      const batchOfRecords = records.slice(batchStartIndex, batchStartIndex + CHANGE_BATCH_SIZE);
+      resolveAfterXChanges(changeChannel, resolve, batchOfRecords.length);
+      changeChannel.publishRecordUpdates(recordType, batchOfRecords, SYNC_QUEUE_KEY);
+    });
+  }
+};
+
+const selectSurveyResponsesForReSync = async db => {
+  const { rows } = await db.runSql(`
+    SELECT survey_response.*
+    FROM survey_response
+    JOIN survey ON survey_response.survey_id = survey.id
+    LEFT JOIN dhis_sync_queue ON survey_response.id = record_id
+    WHERE dhis_sync_queue.id IS NULL`);
+  return rows;
+};
+
+const selectAnswersForReSync = async db => {
+  const { rows } = await db.runSql(`
+    SELECT answer.*
+    FROM answer
+    JOIN survey_response ON answer.survey_response_id = survey_response.id
+    JOIN survey ON survey_response.survey_id = survey.id
+    LEFT JOIN dhis_sync_queue ON answer.id = record_id
+    WHERE
+      survey_response.end_time > '${MIN_SURVEY_END_TIME}'
+      survey.can_repeat = false AND -- repeating surveys are event based, don't need answers in sync queue
+      survey.code NOT IN ('CD3a', 'CD3b') AND -- non org unit entity surveys, event based
+      dhis_sync_queue.id IS NULL;
+ `);
+  return rows;
+};
+
+const updateSyncQueueForSurveyResponses = async db =>
+  db.runSql(`
+    UPDATE dhis_sync_queue
+    SET is_deleted = false, priority = 2 -- use priority 2 not to block up new data syncing through
+    FROM survey_response, survey
+    WHERE
+      record_type = 'survey_response' AND
+      dhis_sync_queue.type = 'update' AND
+      dhis_sync_queue.record_id = survey_response.id AND
+      survey_response.survey_id = survey.id AND
+      survey_response.end_time > '${MIN_SURVEY_END_TIME}'`);
+
+const updateSyncQueueForAnswers = async db =>
+  db.runSql(`
+    UPDATE dhis_sync_queue
+    SET is_deleted = false, priority = 2 -- use priority 2 not to block up new data syncing through
+    FROM answer, survey_response, survey
+    WHERE
+      record_type = 'answer' AND
+      dhis_sync_queue.type = 'update' AND
+      dhis_sync_queue.record_id = answer.id AND
+      answer.survey_response_id = survey_response.id AND
+      survey_response.survey_id = survey.id AND
+      survey_response.end_time > '${MIN_SURVEY_END_TIME}'`);
+
+exports.up = async function (db) {
```

Comment:
This migration is based on [20200218215230-ResyncAllTongaData.js](https://github.com/beyondessential/tupaia/blob/dev/packages/database/src/migrations/20200218215230-ResyncAllTongaData.js)